### PR TITLE
[#ISSUE 305]Add "DeleteRecord" query for deleting records from the backend db

### PR DESCRIPTION
**Pull request by luochen01**
Added a "DeleteRecord" for deleting records from the backup db.

This query has two parameters, i.e., dataset and a seq of filters. The semantics is that records satisfying the filters condition would be deleted from the given data set (basically the same way of SQL)

### CODE REVIEW

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGeneratorTest.scala [hunk 1]
```diff
@@ -695,4 +695,24 @@ class SQLPPGeneratorTest extends Specification {
         """.stripMargin.trim)
     }
   }
+
+  "SQLPPGenerator deleteRecord" should {
+    "generate the delete query without filter" in {
+      val sql = parser.generate(DeleteRecord(TwitterDataSet, Seq.empty), Map(TwitterDataSet -> TwitterDataStore.TwitterSchema))
+      removeEmptyLine(sql) must_== unifyNewLine(
+        """
+          |delete from twitter.ds_tweet t;
```
Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGeneratorTest.scala]: is this statement valid?

Comment (luochen01) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGeneratorTest.scala]: Yes, it's executable by AsterixDB (simply delete everything from the dataset)

Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGeneratorTest.scala]: 😨 it's a scary request. could we ban this kind of without-filter delete request?

Comment (luochen01) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGeneratorTest.scala]: Sure, but it should be fine. This query is not exposed to public, but for internal use only...

File: zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala [hunk 2]
```diff
@@ -97,10 +97,21 @@ class SQLPPGenerator extends AsterixQueryGenerator {
        |)""".stripMargin
   }
 
+  protected def parseDelete(delete: DeleteRecord, schemaMap: Map[String, Schema]): String = {
+    if (delete.filters.isEmpty) {
+      throw new QueryParsingException("Filter condition is required for DeleteRecord query.")
```
Comment (JavierJia) [inline zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala]: 👍

**Comment by haochen07** [comment]
Looks good to me!

**Pull request merged**